### DISCUSSION TITLE
Support for UEFI systems

### DISCUSSION THREAD

**Discussion by pcosman** [oamg/convert2rhel#259] [General]
Most systems use nowadays UEFI instead of the old BIOS boot.
So, in this context I assume a huge amount of Centos users are eagerly awaiting a version of convert2rhel that supports UEFI systems. I've read here on GitHub that there is currently work underway to support UEFI. Is there any estimated arrival date for this update of convert2rhel ?

**Comment by bocekm** [comment]
Hi @pcosman. We understand the need and demand to allow conversions of UEFI based systems. However our immediate priority now is to start supporting conversions of Oracle Linux to RHEL. That comes with the need for an update to our CI. We will get to the UEFI support right after that. There's unfortunately no specific date I could promise to you.

**Reply by pcosman** [reply]
Would using version 0.17 be an option? Knowing about the UEFI bug and following the resolutions for that bug that can be found online?

**Reply by bocekm** [reply]
You can try that out at your own risk if you have a full system backup, as an option not supported by Red Hat.

Another unsupported option, but better from my POV given the number of bug fixes added in versions 0.18 through 0.20, would be to install the 0.20 version and change manually `critical` to `warn` at line https://github.com/oamg/convert2rhel/blob/v0.20/convert2rhel/checks.py#L48. The path to this file depends on the major version of CentOS you plan to convert, but for example on CentOS 8 it's `/usr/lib/python3.6/site-packages/convert2rhel/checks.py`.

**Reply by Venefilyn** [reply]
Hi @trivore

Thanks for the info. Glad you were able to convert! 

We haven't delved deeper into what if anything needs to change to support secureboot without issues. So any conversion that circumvents the check will be unsupported.

We might create a way to get around the secureboot check, however if one does this it will also be unsupported
